src/services/pool/concerns/apr.concern.ts

```diff
@@ -0,0 +1,178 @@
+import { isStablePhantom } from '@/composables/usePool';
+import { FiatCurrency } from '@/constants/currency';
+import { bnSum, bnum } from '@/lib/utils';
+import { calcUSDPlusWeightedAPR } from '@/lib/utils/apr.helper';
+import { includesWstEth } from '@/lib/utils/balancer/lido';
+import { aaveService } from '@/services/aave/aave.service';
+import { AprRange, Pool, PoolAPRs } from '@/services/balancer/subgraph/types';
+import { TokenPrices } from '@/services/coingecko/api/price.service';
+import { lidoService } from '@/services/lido/lido.service';
+import { GaugeBalApr } from '@/services/staking/staking-rewards.service';
+
+export class AprConcern {
+  constructor(
+    public pool: Pool,
+    public readonly lido = lidoService,
+    public readonly aave = aaveService
+  ) {}
+
+  public async calc(
+    poolSnapshot: Pool | undefined,
+    prices: TokenPrices,
+    currency: FiatCurrency,
+    protocolFeePercentage: number,
+    stakingBalApr: GaugeBalApr,
+    stakingRewardApr = '0'
+  ): Promise<PoolAPRs> {
+    const swapFeeAPR = this.calcSwapFeeAPR(poolSnapshot, protocolFeePercentage);
+    const yieldAPR = await this.calcYieldAPR(
+      prices,
+      currency,
+      protocolFeePercentage
+    );
+    const unstakedTotalAPR = bnSum([swapFeeAPR, yieldAPR.total]).toString();
+    const aprGivenBoost = (boost = '1') =>
+      this.calcAprGivenBoost(
+        unstakedTotalAPR,
+        stakingBalApr,
+        stakingRewardApr,
+        boost
+      );
+    const stakedAprRange = this.calcStakedAprRange(
+      unstakedTotalAPR,
+      stakingBalApr,
+      stakingRewardApr
+    );
+
+    return {
+      total: {
+        unstaked: unstakedTotalAPR,
+        staked: {
+          calc: aprGivenBoost,
+          ...stakedAprRange
+        }
+      },
+      swap: swapFeeAPR,
+      yield: yieldAPR,
+      staking: {
+        bal: stakingBalApr,
```

Comment:
Is this only for the veBAL pool?

Reply:
No, this is the BAL emissions APR for any pool that receives them.